frontend/src/Event/MeetEventDisplay.jsx

```diff
@@ -74,6 +75,20 @@ const MeetEventDisplay = () => {
   const handleRankingClick = (id) => {
     console.log("Ranking ...");
   };
+  const handleDownloadDetails = async (id) => {
+    setSelectedEventIndex(Number(id));
```

Comment:
`selectedEventIndex` should be set only if it's null (when clicking on the event table).
Setting it again on download click causes a rerender of `MeetEventDisplay`, which returns the view to the event table.